index.js

```diff
@@ -1,30 +1,118 @@
 /* jshint node: true */
+/* jshint esnext: false */
 'use strict';
 
+var Funnel = require('broccoli-funnel');
+var mergeTrees = require('broccoli-merge-trees');
 var path = require('path');
+var rename = require('broccoli-stew').rename;
+var AMDDefineFilter = require('./lib/amd-define-filter');
+var fs = require('fs');
+
+function lookupPackage(packageName) {
+  var modulePath = require.resolve(packageName);
+  var i = modulePath.lastIndexOf(path.sep + 'build');
+  return modulePath.slice(0, i);
+}
 
 module.exports = {
   name: 'ember-d3',
 
-  included: function(app) {
-    this._super.included(app);
-    if (process.env.EMBER_CLI_FASTBOOT !== 'true') {
+  isDevelopingAddon: function () {
+    return false;
+  },
+
+  d3Modules: [
+
+    // Imported from package.json
+  ],
+
+  /**
+   * `import()` taken from ember-cli 2.7 to allow nested addon usage.
+   */
+  import: function (asset, options) {
+    var app = this.app;
+    while (app.app) {
+      app = app.app;
+    }
+
+    app.import(asset, options);
+  },
 
-      // Fix for loading it in addons/engines
-      if (typeof app.import !== 'function' && app.app) {
-        app = app.app;
+  included: function (app) {
+    this._super.included && this._super.included.apply(this, arguments);
+    this.app = app;
+
+    while (app.app) {
+      app = app.app;
+    }
+
+    var pkg = require(path.join(lookupPackage('d3'), 'package.json'));
+
+    // Find all dependencies of `d3`
+    this.d3Modules = Object.keys(pkg.dependencies).filter(function (name) {
+      return /^d3\-/.test(name);
+    });
+
+    // This essentially means we'll skip importing this package twice, if it's
+    // a dependency of another package.
+    if (!app.import) {
+      if (this.isDevelopingAddon()) {
+        console.log('[ember-cli-d3-shape] skipping included hook for', app.name);
```

Comment:
This may break on older versions of node, which don't have a `console` global. Prefer to use 
https://ember-cli.com/api/classes/UI.html#method_write